p2p/discover/common.go

```diff
@@ -34,8 +35,8 @@ import (
 
 // UDPConn is a network connection on which discovery can operate.
 type UDPConn interface {
-	ReadFromUDP(b []byte) (n int, addr *net.UDPAddr, err error)
-	WriteToUDP(b []byte, addr *net.UDPAddr) (n int, err error)
+	ReadFromUDPAddrPort(b []byte) (n int, addr netip.AddrPort, err error)
+	WriteToUDPAddrPort(b []byte, addr netip.AddrPort) (n int, err error)
```

Comment:
Isn't the typing self-explanatory here? I would even go as far to say `Read(..)` and `Write(..)` would be okay in this context since the interface says we're dealing with UDP and the parameters specify already `netip.AddrPort`. Is there another reason for this naming?

Reply:
Ah okay I see!